CMakePresets.json

```diff
@@ -0,0 +1,172 @@
+{
+	"version": 3,
+	"cmakeMinimumRequired": {
+		"major": 3,
+		"minor": 19,
+		"patch": 0
+	},
+	"configurePresets": [
+		{
+			"name": "debug-flags",
+			"hidden": true,
+			"cacheVariables": {
+				"CMAKE_BUILD_TYPE": "Debug",
+				"CMAKE_C_FLAGS": "-fno-omit-frame-pointer -fno-optimize-sibling-calls -gfull -O0",
+				"CMAKE_CXX_FLAGS": "-fno-omit-frame-pointer -fno-optimize-sibling-calls -gfull -O0"
+			},
+      "environment": {
+        "VCPKG_SUFFIX": ""
+      }
+		},
+		{
+			"name": "release-flags",
+			"hidden": true,
+			"cacheVariables": {
+				"CMAKE_BUILD_TYPE": "Release"
+			},
+      "environment": {
+        "VCPKG_SUFFIX": "-rel"
+      }
+		},
+    {
+      "name": "arm64",
+      "hidden": true,
+      "environment": {
+        "VCPKG_ARCH": "arm64"
+      },
+      "architecture": {
+        "value": "arm64",
+        "strategy": "external"
+      }
+    },
+    {
+      "name": "x86_64",
+      "hidden": true,
+      "environment": {
+        "VCPKG_ARCH": "x64"
+      },
+      "architecture": {
+        "value": "x64",
+        "strategy": "external"
+      }
+    },
+		{
+			"name": "linux",
+			"hidden": true,
+			"cacheVariables": {
+				"CMAKE_C_COMPILER": "$env{VCPKG_ROOT}/installed/x64-linux$env{VCPKG_SUFFIX}/tools/llvm/clang",
+				"CMAKE_CXX_COMPILER": "$env{VCPKG_ROOT}/installed/x64-linux$env{VCPKG_SUFFIX}/tools/llvm/clang++"
```

Comment:
I'm not sure we really want to rely on the compiler built by vcpkg for building the tool itself. Just use whatever CMake finds.

Reply:
So I personally really like picking the specific compiler from vcpkg! I found that for me, the default cmake always makes a *very poor* decision regarding picking a compiler, and then I have to remember to delete my build and re-set `CC` & `CXX`.

Is there a reason we would not want to default to the cxx-common compiler, given that we already ship a working compiler anyway? Afaik we can always override CMakePresets with explicit directives to CMake.

Reply:
I know there have been some previous issues with the vcpkg-built compiler not being able to compile with an arbitrary sanitizer enabled, and I haven't tested it recently, but it sounds like a simple compile is currently working.

Yes, setting a different compiler for CMake on the CLI will work. 👍